File: Library.HospitalSystem/Models/Appointment.cs
using System;

namespace Library.HospitalSystem.Models;

public class Appointment
{
    public uint Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public DateTime DateTimeStart { get; set; }
    public DateTime DateTimeEnd { get; set; }
    public uint PhysicianId { get; set; }
    public uint PatientId { get; set; }
    public Appointment()
    {
        Id = 0;
        Title = string.Empty;
        Description = string.Empty;
        DateTimeStart = DateTime.MinValue;
        DateTimeEnd = DateTime.MinValue;
        PhysicianId = 0;
        PatientId = 0;
    }
    public override string ToString()
    {
        return $"{Id} {Title} Start: {DateTimeStart} End: {DateTimeEnd} Physician Id: {PhysicianId} Patient Id: {PatientId}";
    }
}

File: next-frontend-hospital-system/src/types/api/appointment.ts
import { PatientDTO } from "./patient";
import { PhysicianDto } from "./physician";

export interface AppointmentDto {
    id: number;
    title: string;
    description: string;
    dateTimeStart: string;
    dateTimeEnd: string;
    physicianId: number;
    patientId: number;
    billId: number;
    patientNav: PatientDTO;
    physicianNav: PhysicianDto;
    billNav: BillDto;
    appointmentTreatmentsNav: AppointmentTreatment[];
}

export interface BillDto {
    id: number;
    amount: number;
    outOfPocket: number;
}

export interface Treatment {
    id: number;
    name: string;
    price: number;
}

export interface AppointmentTreatment {
    id: number;
    appointmentId: number;
    treatmentId: number;
    treatment: Treatment;
}

export interface AppointmentWithNavDto {
    id: number;
    title: string;
    description: string;
    dateTimeStart: string;
    dateTimeEnd: string;
    physicianId: number;
    patientId: number;
    patientNav: PatientDTO;
    physicianNav: PhysicianDto;
}

export interface AppointmentCreateDto {
    title: string;
    description: string;
    dateTimeStart: string;
    dateTimeEnd: string;
    physicianId: number;
    patientId: number;
    treatmentOptionIds: string[];
}
